### PR TITLE
GA-40: Reusable schema validators (CIDR, enum, port range)

### LINKED ISSUE #52: GA-40: Reusable schema validators (CIDR, enum, port range)
## Background

Currently no schema validators exist in the provider. Invalid values (e.g. a non-CIDR string as `cidr.ipv4`, a port number of 70000, an unknown `direction` value) are silently accepted at plan time and fail at API time with cryptic errors. Validators surface these issues immediately with human-readable messages.

## Requirements

Implement the following reusable validators in `internal/provider/validators.go`:

1. **CIDR block validator** — validates that a string is a valid IPv4 or IPv6 CIDR block (e.g. `10.100.0.0/16`). Use `net.ParseCIDR` from the Go standard library.

2. **String enum validator** — validates that a string value is one of a fixed set of allowed values. Reusable for `direction` (ingress/egress), `version` (IPv4/IPv6), `protocol` (tcp/udp/icmp/any), etc.

3. **Port range validator** — validates that an integer value is in range 1–65535.

Apply validators to existing resources once implemented:
- `seca_block_storage`: none currently needed.
- `seca_network.cidr.ipv4`: CIDR validator.
- `seca_internet_gateway`: none.
- `seca_route_table.routes[].destination_cidr_block`: CIDR validator.
- `seca_subnet.cidr.ipv4`: CIDR validator.
- `seca_security_group.rules[].direction`: enum validator (`ingress`, `egress`).
- `seca_security_group.rules[].ports.list[]`: port range validator (1–65535).
- `seca_security_group.rules[].ports.from`: port range validator.
- `seca_security_group.rules[].ports.to`: port range validator.
- `seca_public_ip.version`: enum validator (`IPv4`, `IPv6`).
- `seca_role.permissions[].verb[]`: enum validator (verify allowed verbs from API).

## Implementation Notes

- File: `internal/provider/validators.go`
- Use `github.com/hashicorp/terraform-plugin-framework-validators` if available in `go.mod`, or implement via `validator.String` / `validator.Int64` interfaces.
- Document validators in `ai/provider-conventions.md`.

## Acceptance Criteria

- [ ] `terraform plan` with an invalid CIDR emits a clear validation error, not an API error.
- [ ] `terraform plan` with `direction = "INVALID"` emits a clear validation error.
- [ ] `terraform plan` with port `70000` emits a clear validation error.
- [ ] All validators have unit tests covering valid and invalid inputs.

## Testing Requirements

- Unit tests: `validators_test.go` — test each validator with valid values, boundary values, and invalid values.
- No new acceptance tests required (validators are exercised at plan time).

## Documentation Requirements

- Document validators in `ai/provider-conventions.md` Validators section.

## Dependencies

- GA-16 (route_table), GA-18 (subnet), GA-21 (security_group), GA-23 (public_ip): apply validators when these resources land.

## Estimated Complexity

M

## Definition of Done

- All validators implemented with unit tests. Applied to all applicable schema attributes. Documented in `ai/provider-conventions.md`.

### PR DESCRIPTION
**Pull request by aru-amedeo**
Closes #52

### CODE REVIEW

File: internal/provider/resource_network.go [hunk 1]
```diff
@@ -77,8 +78,9 @@ type NetworkResourceModel struct {
 func (r *NetworkResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
```
Comment (jannotti-ionos) [inline internal/provider/resource_network.go:78]: Will not implement these kinds of validations in the other resources with field validations ?
https://github.com/search?q=repo%3Aeu-sovereign-cloud%2Fspec+x-kubebuilder-validation&type=code

Comment (aru-amedeo) [reply internal/provider/resource_network.go:78]: Yes, I applied validators to all resources that have matching fields.

The SDK spec distinguishes the two via CEL rules (`cidr(self).ip().family() == 4/6`), so the generic validator would have silently accepted an IPv4 address in the IPv6 field and vice versa. The generic `CIDRValidator` is kept for `destination_cidr_block` in the route table, which accepts both families.

**`resource_security_group.go`:** `direction` gets `StringEnumValidator("ingress", "egress")`, `protocol` gets `StringEnumValidator("tcp", "udp", "tcp+udp", "icmp")` (from the SDK's `x-kubebuilder-validation-enum` tags), and `ports.from`/`ports.to` get `PortRangeValidator()`. List element validation for `ports.list` would need `terraform-plugin-framework-validators` (not yet a dep); tracked separately.

Branch was rebased onto main to pick up the subnet and security group resources.

**Review by jannotti-glaucio — APPROVED** [review]

**Review by ghost — APPROVED** [review]
